src/test/java/org/mockito/internal/configuration/plugins/PluginLoaderTest.java

```diff
@@ -0,0 +1,87 @@
+package org.mockito.internal.configuration.plugins;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.internal.util.io.IOUtil;
+import org.mockitoutil.ClassLoaders;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test PluginLoader. This test check that loaded plugin can be assigned to fields of classes what loaded by
+ * application class loader
+ */
+public class PluginLoaderTest {
+
+    public static final String PLUGIN_PATH = "mockito-extensions/org.mockito.internal.configuration.plugins.PluginInterface";
+    private PluginLoader pluginLoader;
+
+    @Before
+    public void setUp() {
+        pluginLoader = new PluginLoader(new DefaultPluginSwitch());
+    }
+
+    @Test
+    public void should_load_custom_plugin_without_context_classloader() {
+
+        Object pluginInterface = loadPlugin();
+
+        assertThatLoadedPluginAssignable(pluginInterface);
+    }
+
+
+    @Test
+    public void should_load_custom_plugin_with_custom_context_classloader() throws Exception {
+
+        ClassLoader classLoader = ClassLoaders.inMemoryClassLoader().withParent(getParentClassLoader())
+                                          .withClassDefinition("", readResource()).build();
```

Comment:
If I remove `.withClassDefinition("", readResource())` then test will fail, because default plugin implementer will be loaded. Without class definition the in-memory classloader returns empty resource `Enumeration` . So I need a easy way to create class loader which will return as resource that I need to return and failed to load any class. Any suggestion?